src/Sulu/Bundle/Sales/OrderBundle/Resources/public/js/components/orders/components/form/main.js

```diff
@@ -289,21 +424,135 @@ define([
         },
 
         /**
-         * called when headerbar should be saveable
+         * Called when contact auto-complete is changed.
+         *
+         * @param {Object} contact
+         */
+        contactChangedListener = function(contact) {
+            var id = (contact && contact.id) ? contact.id : null;
+
+            // If contact has been changed.
+            if (id !== this.contactId) {
+                this.contactId = id;
+
+                if (id) {
+                    initSelectsByContact.call(this, contact);
+                } else {
+                    initContactSelect.call(this, []);
+                    initAddressComponents.call(this, [], constants.deliveryAddressInstanceName);
+                    initAddressComponents.call(this, [], constants.billingAddressInstanceName);
+
+                    setSettingsOverlayAdresses.call(this, []);
+                }
+            }
+        },
+
+        /**
+         * @param {Number} customerTypeId
+         */
+        customerTypeChangedHandler = function(customerTypeId) {
+            this.customerTypeId = customerTypeId;
+
+            switch (customerTypeId) {
+                case CUSTOMER_TYPE.ORGANIZATION:
+                    startCustomerAccountAutocomplete.call(this);
+                    accountChangedListener.call(this, null);
+                    break;
+                case CUSTOMER_TYPE.PRIVAT_PERSON:
+                    startCustomerContactAutocomplete.call(this);
+                    contactChangedListener.call(this, null);
+                    break;
+            }
+        },
+
+        /**
+         * Called when headerbar should be saveable.
          */
         changeHandler = function() {
             setSaved.call(this, false);
         },
 
         /**
-         * sets dependent selects based on currently selected account
-         * @param id
-         * @param orderData
+         * @param {Object} contact
+         * @param {Object} orderData
+         */
+        initSelectsByContact = function(contact, orderData) {
+            var data = null;
+            var preselect = null;
+
+            // Load contact from repository.
+            var contactRepository = Contact.findOrCreate({id: contact.id});
+            contactRepository.fetch({
+                success: function(model) {
+                    var contactModel = model.toJSON();
```

Comment:
does that code differ from account? try to use the same callback function for both account and contact